web/third/bootstrap-typeahead/typeahead.js

```diff
@@ -313,12 +317,18 @@ import {get_string_diff} from "../../src/util";
       })
     }
 
-  , render: function (items) {
+  , render: function (matching_items, total_items) {
       var that = this
 
-      items = $(items).map(function (i, item) {
-        i = $(that.options.item).data('typeahead-value', item)
-        i.find('a').html(that.highlighter(item))
+      const items = $(matching_items).map((index, item) => {
+        const i = $(that.options.item).data('typeahead-value', item)
+        const item_html = i.find('a').html(that.highlighter(item))
+
+        const option_label_html = that.option_label(total_items, item)
+
+        if (option_label_html) {
+          item_html.append(option_label_html).addClass("new-topic-typeahead");
+        }
```

Comment:
Rename this class to something like `typeahead_option_label`. Of course this should be reflected in the CSS file as well

Reply:
LOL, I'm so confused now. But in the previous comment you said:
> I had a closer look at the code and I think `items` -> `total_items` (or maybe even final_items) and `item_list` -> `matching_items` would be fine. What do you think?

I think that makes sense. With the new suggestion I think you got the `sort` function wrong. Before sorting, the array will contain all the topics in the streams. After sorting the array, it will only contain 3 closest matching topics to the query and the topic itself. So I would think that the current names are fine with the before sorting being `total_items` and then after sorting being `matching_items` since we'll be using them in `render`.

Are we sure we should still change it up 🤔

Reply:
Hmm, these names can be interpreted differently and hence the confusion probably... Sorry! 😬 I'll try and (hopefully) explain better why we should switch.

`total_items` is probably a bad naming choice, `final_items` conveys the meaning that I was going for... the final list of options that are displayed including the new topic. `matching_items` I'd thought as indicating options that have actually been found by matching, unlike the new option that is tacked on later. So to sum up: `matching_items` on `sort`ing should return `final_items`. Does that make sense?

Reply:
Sorry, you're right after sorting the array it will contain all the topics and the query itself, but when we pass it into the `render` function `matching_items` only contains 3 closest matching topics to the query and the topic itself. I feel like it makes more sense with the way it is now.

Reply:
Hmm yeah I think that makes more sense now. Could we continue to use matching_items in the `render` function then?
`, render: function (matching_items, total_items) {`

Reply:
We should change the names everywhere for consistency in my opinion. Simply replacing all instances of `matching_items` with `final_items` and then replacing all instances of `total_items` with `matching_items` should be perfect